util/src/main/java/org/apollo/util/security/UserStats.java

```diff
@@ -0,0 +1,371 @@
+package org.apollo.util.security;
+
+import io.netty.buffer.ByteBuf;
+import org.apollo.util.BufferUtil;
+
+public class UserStats {
+
+	/**
+	 * Current version of the {@Link UserStats}.
+	 */
+	public static final int STATS_VERSION = 8;
+
+	/**
+	 * The reported version.
+	 */
+	private final short version;
+
+	/**
+	 * The type of operating system.
+	 */
+	private final int osType;
+	/**
+	 * If the operating system uses 64bits.
+	 */
+	private final boolean os64Bit;
+	/**
+	 * The version of the OS.
+	 */
+	private final int osVersion;
+	/**
+	 * The vendor of the JVM.
+	 */
+	private final int javaVendor;
+	/**
+	 * The major version of the active JVM.
+	 */
+	private final int javaMajor;
+	/**
+	 * The minor version of the active JVM.
+	 */
+	private final int javaMinor;
+	/**
+	 * The build of the active JVM.
+	 */
+	private final int javaBuild;
+	/**
+	 * Number of processors available.
+	 */
+	private final int availableProcessors;
+	/**
+	 * Driver release year.
+	 */
+	private final int gpuDriverYear;
+	/**
+	 * Driver release month.
+	 */
+	private final int gpuDriverMonth;
+	/**
+	 * If this is running off a console.
+	 */
+	private final boolean isConsole;
+	/**
+	 * Maximum heap memory.
+	 */
+	private final int maxMemory;
+	/**
+	 * The number of CPUs on the machine.
+	 */
+	private final int cpuCount;
+	/**
+	 * The number of RAM available on the machine.
+	 */
+	private final int ramAmount;
+	/**
+	 * The current CPU's clock speed.
+	 */
+	private final int cpuClockspeed;
+	/**
+	 * The description of the gpu's driver.
+	 */
+	private final String gpuDriverDescription;
+	/**
+	 * The name of the GPU.
+	 */
+	private final String gpuName;
+	/**
+	 * The directX version installed.
+	 */
+	private final String directxVersion;
+	/**
+	 * The CPU's brand identifier.
+	 */
+	private final int cpuBrandId;
+	/**
+	 * The CPU"s vendor identifier.
+	 */
+	private final String cpuVendor;
+	/**
+	 * The cpu brand's name.
+	 */
+	private final String cpuBrandString;
+	/**
+	 * Features on the CPU.
+	 */
+	private final int[] cpuFeatures = new int[3];
+	/**
+	 * The model of the CPU.
+	 */
+	private final int cpuModel;
+
+	/**
+	 * Instantiates a new User stats.
+	 *
+	 * @param buffer the buffer
+	 */
+	public UserStats(ByteBuf buffer) {
```

Comment:
A constructor isn't the place to be reading from a byte buffer. Create a static constructor and make it clear that the buffer is consumed.